### ISSUE TITLE
Benchmark to print out device information

### ISSUE THREAD

**Issue by kknox** [clMathLibraries/clSPARSE#38]
The benchmark apps need to print out information about the device under test.   Some machines can have multiple devices installed, and the benchmark does not inform the user which device it picked as the default device.

Related to this, the benchmark needs to provide a facility to print out a list of all detected platforms and devices.  This will enable users to be able to select the device they wish as described in #39

**Comment by jpola** [comment]
We have some simple info about the device. It is very simple but it shows which device is used. On the other hand the code haven't got the functionality to pick the particular device.

I vote for closing this one and opening the right enhancement issue describing the need for such functionality.

**Comment by kknox** [comment]
Yes, I would eventually like to be able to use our bench and test programs as a substitute for clinfo.  clinfo only comes with the APP SDK, and it would be nice to be able to query what opencl devices/properties are found no matter which SDK is used.  It's helpful to us when end-users report problems.
